common/postgresql-ng/templates/deployment.yaml

```
@@ -90,8 +91,13 @@ spec:
         - name: USER_PASSWORD_{{ $user | replace "-" "_" }}
           valueFrom:
             secretKeyRef:
+        {{ $settings := $settings | default (dict) }}{{- if not (and (hasKey $settings "secretName") (hasKey $settings "secretKey")) }}
```

```suggestion
        {{- $settings := $settings | default (dict) }}{{ if not (and (hasKey $settings "secretName") (hasKey $settings "secretKey")) }}
```

Also, if there is a `not` at the top level of the `if`, it usually reads better to flip the branches.